okhttp-apache/src/main/java/com/squareup/okhttp/apache/OkApacheClient.java

```diff
@@ -39,12 +41,22 @@
  * and others.
  */
 public final class OkApacheClient implements HttpClient {
-  private static Request transformRequest(HttpRequest request) {
+  private static Request transformRequest(HttpHost httpHost, HttpRequest request) {
     Request.Builder builder = new Request.Builder();
 
     RequestLine requestLine = request.getRequestLine();
     String method = requestLine.getMethod();
-    builder.url(requestLine.getUri());
+    String url = requestLine.getUri();
+    if (httpHost != null) {
+      try {
+        URI uri = URI.create(url);
+        URI uriWithHost = new URI(httpHost.getSchemeName(), uri.getUserInfo(), httpHost.getHostName(),
+            httpHost.getPort(), uri.getPath(), uri.getQuery(), uri.getFragment());
+        url = uriWithHost.toString();
+      } catch (URISyntaxException ignored) {
```

Comment:
I think we want to re-throw an exception here. Maybe an IllegalArgumentException? Otherwise the HttpHost will be ignored if it's malformed, hiding bugs.